offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -277,14 +279,16 @@ class Offer {
     }
 
     private BigDecimal calculateItemNet(ProductItem item) {
-        BigDecimal unitPrice = selectedCustomerAffiliation.category == AffiliationCategory.INTERNAL ? item.product.internalUnitPrice : item.product.externalUnitPrice
-        return unitPrice * item.quantity
+        return getUnitPrice(item) * item.quantity
     }
 
     private double calculateItemOverhead(ProductItem item) {
         return (calculateItemNet(item) - item.quantityDiscount) * overhead
     }
 
+    private BigDecimal getUnitPrice(ProductItem item){
+        return selectedCustomerAffiliation.category == AffiliationCategory.INTERNAL ? item.product.internalUnitPrice : item.product.externalUnitPrice
```

Comment:
You should use the appropriate BigDecimal factory method. Otherwise you cannot guarantee that the casting was successful. `new BigDecimal(double)` does not lead to the expected behaviour. I do not know which casting method groovy applies here.

Reply:
`BigDecimal.valueOf()` should be used instead, I think.